pallets/automation-time/src/tests.rs

```diff
@@ -723,6 +724,91 @@ fn force_cancel_task_works() {
 	})
 }
 
+mod run_dynamic_dispatch_action {
+	use super::*;
+	use sp_runtime::DispatchError;
+
+	#[test]
+	fn cannot_decode() {
+		new_test_ext(START_BLOCK_TIME).execute_with(|| {
+			let account_id = AccountId32::new(ALICE);
+			let task_id = AutomationTime::generate_task_id(account_id.clone(), vec![1]);
+			let bad_encoded_call: Vec<u8> = vec![1];
+
+			let weight = AutomationTime::run_dynamic_dispatch_action(
+				account_id.clone(),
+				bad_encoded_call,
+				task_id,
+				false,
+			);
+
+			assert_eq!(
+				weight,
+				<Test as Config>::WeightInfo::run_dynamic_dispatch_action_fail_decode()
+			);
+			assert_eq!(
+				events(),
+				[Event::AutomationTime(crate::Event::CallCannotBeDecoded {
+					who: account_id,
+					task_id,
+				}),]
+			);
+		})
+	}
+
+	#[test]
+	fn call_errors() {
+		new_test_ext(START_BLOCK_TIME).execute_with(|| {
+			let account_id = AccountId32::new(ALICE);
+			let task_id = AutomationTime::generate_task_id(account_id.clone(), vec![1]);
+			let call: Call = frame_system::Call::set_code { code: vec![] }.into();
+			let encoded_call = call.encode();
+
+			AutomationTime::run_dynamic_dispatch_action(
+				account_id.clone(),
+				encoded_call,
+				task_id,
+				false,
+			);
+
+			assert_eq!(
+				events(),
+				[Event::AutomationTime(crate::Event::DynamicDispatchResult {
+					who: account_id,
+					task_id,
+					result: Err(DispatchError::BadOrigin),
+				}),]
+			);
+		})
+	}
+
+	#[test]
+	fn call_works() {
+		new_test_ext(START_BLOCK_TIME).execute_with(|| {
+			let account_id = AccountId32::new(ALICE);
+			let task_id = AutomationTime::generate_task_id(account_id.clone(), vec![1]);
+			let call: Call = frame_system::Call::remark { remark: vec![] }.into();
+			let encoded_call = call.encode();
+
+			AutomationTime::run_dynamic_dispatch_action(
+				account_id.clone(),
+				encoded_call,
+				task_id,
+				false,
+			);
+
+			assert_eq!(
+				events(),
+				[Event::AutomationTime(crate::Event::DynamicDispatchResult {
```

Comment:
I expected to see the remark event here as well. What am I missing?

Reply:
for reasons I will never understand there is `frame_system::Call::remark ` and `frame_system::Call::remarkWithEvent`. I have no idea what value of the first is but it's what is being used here so no event is fired.

Reply:
Oh! Well that atleast explains my confusion.

What do you think of using the `frame_system::Call::remark` for benchmarking rather than the `benchmark: bool` arg I suggested earlier? `frame_system::Call::remark` has a constant and very small weight and seems like a much cleaner way to do the benchmarking.